core/ledger/snapshotgrpc/snapshot_service_test.go

```diff
@@ -87,37 +96,49 @@ func TestSnapshot(t *testing.T) {
 			errMsg:        "failed to unmarshal snapshot request: proto: can't skip unknown wire type 4",
 		},
 		{
-			name:      "missing channel ID",
-			channelID: "",
-			errMsg:    "missing channel ID",
+			name:          "missing channel ID",
+			channelID:     "",
+			signedRequest: createSignedQuery(""),
+			errMsg:        "missing channel ID",
+		},
+		{
+			name:          "missing signature header",
+			channelID:     "",
+			signedRequest: &pb.SignedSnapshotRequest{Request: protoutil.MarshalOrPanic(&pb.SnapshotQuery{})},
+			errMsg:        "missing signature header",
+		},
+		{
+			name:          "cannot find ledger",
+			channelID:     ledgerID,
+			signedRequest: createSignedQuery(ledgerID),
+			errMsg:        "cannot find ledger for channel " + ledgerID,
 		},
 		{
-			name:      "cannot find ledger",
-			channelID: ledgerID,
-			errMsg:    "cannot find ledger for channel " + ledgerID,
+			name:          "checkACL error propagation",
+			channelID:     ledgerID,
+			signedRequest: createSignedQuery(ledgerID),
+			checkACLErr:   fmt.Errorf("fake-check-acl-error"),
+			errMsg:        "fake-check-acl-error",
```

Comment:
I found this aclprovider error test a bit of a force fit into the existing test case - definition. I feel that having this as independent one makes a better sense.